fix incorrect setting of $ac_cv_sys_symbol_underscore.
There is no reason not to AC_COMPILE and use the set cross-nm to look for
the symbol mangling scheme. Thus it is incorrect to check for cross compilation
in the GNUPG_SYS_SYMBOL_UNDERSCORE check.

Origin:

aldot at uclibc.org aldot at uclibc.org
Wed Sep 19 12:20:53 UTC 2007

http://lists.uclibc.org/pipermail/buildroot/2007-September/017322.html

--- a/acinclude.m4
+++ b/acinclude.m4
@@ -76,13 +76,14 @@ case "${host}" in
     i386-emx-os2 | i[3456]86-pc-os2*emx | i386-pc-msdosdjgpp)
         ac_cv_sys_symbol_underscore=yes
         ;;
-    *)
-      if test "$cross_compiling" = yes; then
-        if test "x$ac_cv_sys_symbol_underscore" = x ; then
-           ac_cv_sys_symbol_underscore=yes
-        fi
-      else
-         tmp_do_check="yes"
+    *) if test -z "$ac_cv_sys_symbol_underscore"; then
+         if test "$cross_compiling" = yes; then
+           if test "x$ac_cv_sys_symbol_underscore" = x ; then
+              ac_cv_sys_symbol_underscore=yes
+           fi
+         else
+            tmp_do_check="yes"
+         fi
       fi
        ;;
 esac
 
